flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricFetcherTest.java

```diff
@@ -263,7 +220,151 @@ public void testShortUpdateInterval() throws InterruptedException {
+                                                                    requestMetricsAnswer);
+                                                        });
+                                        return metricsAnswerFuture;
+                                    } else {
+                                        return CompletableFuture.completedFuture(
+                                                requestMetricsAnswer);
+                                    }
+                                })
+                        .build();
+
+        // ========= setup JobManager
+        // ==================================================================================
+
+        final TestingRestfulGateway restfulGateway =
+                new TestingRestfulGateway.Builder()
+                        .setRequestMultipleJobDetailsSupplier(
+                                () ->
+                                        CompletableFuture.completedFuture(
+                                                new MultipleJobsDetails(Collections.emptyList())))
+                        .setRequestMetricQueryServiceGatewaysSupplier(
+                                () -> {
+                                    if (requestMetricQueryServiceGatewaysCounter != null) {
+                                        requestMetricQueryServiceGatewaysCounter.incrementAndGet();
+                                    }
+                                    return CompletableFuture.completedFuture(
+                                            Collections.singleton(jmQueryService.getAddress()));
+                                })
+                        .setRequestTaskManagerMetricQueryServiceGatewaysSupplier(
+                                () ->
+                                        CompletableFuture.completedFuture(
+                                                Collections.singleton(
+                                                        Tuple2.of(
+                                                                tmRID,
+                                                                tmQueryService.getAddress()))))
+                        .build();
+
+        final GatewayRetriever<RestfulGateway> retriever =
+                () -> CompletableFuture.completedFuture(restfulGateway);
+
+        // ========= start MetricFetcher testing
+        // =======================================================================
+        return new MetricFetcherImpl<>(
+                retriever,
+                address -> CompletableFuture.completedFuture(tmQueryService),
+                Executors.directExecutor(),
+                timeout,
+                updateInterval);
+    }
+
+    private static Runnable waitTimeMs(long sleepTimeMs) {
+        return () -> {
+            try {
+                Thread.sleep(sleepTimeMs);
+            } catch (Throwable throwable) {
+                fail(throwable.getMessage());
```

Comment:
Maybe just throw an exception directly is ok.

Reply:
I have removed the method because only one situation used it.